### DISCUSSION TITLE
Paper writing

### DISCUSSION THREAD

**Discussion by Muennighoff** [embeddings-benchmark/mteb#595] [General]
Should we already create an Overleaf & get started on writing? I have Overleaf premium so created one here: https://www.overleaf.com/8693731561tpntkddrhngs#41bcff (but if someone else wants to own it, feel free to recreate)
It may make sense to ask every contributor to introduce their dataset in a paragraph in the Appendix?

NeurIPS D&B would be great I think but DDL is in 1 month - do you think we can make it @KennethEnevoldsen @imenelydiaker?

**Comment by imenelydiaker** [comment]
Thanks for creating the Overleaf, we should definitely start writing!

For the conferences, there is also the next ARR cycle for *CL conferences, this one would give us access to EMNLP, the deadline is June 15th.

The deadline for end of contributions is May 15th, I think we're still missing some languages. We can post about the missing languages and change the deadline for end of contributions, either way I think we need to start writing now else we'll be too short for both deadlines.

**Reply by KennethEnevoldsen** [reply]
Sounds like a good idea.

We should probably create the outline for the structure.

Here is my proposed structure:


Abstract
- primary contributions:
  - notable speedups of MTEB
  - coverage of more than X languages while maintaining evaluation speed
  - broader coverage of domains, languages, code and tasks

Introduction
- Limitations of MTEB
- Contributions

Introduction/related work
- speeding up benchmarks
- large collaborative effort in NLP

Methods

Methods/ensuring dataset quality
- metadata (non-translated datasets), non-trivial performance

Methods/constructing benchmarks
- Introduce benchmarks
  - We define a set of benchmarks, and introduce the notation of these:
    - Monolingual benchmarks: MMTEB(eng), MMTEB(fra), etc. 
    - Code benchmarks: MMTEB(code)
    - Multilingual benchmarks: MMTEB(multilingual), MMTEB(indo-euro), etc.
- We might also discuss how we select tasks + aggregate scores

Methods/Speeding up the benchmark:
- to make the benchmark more accessible to low resource communities
- to encourage broader evaluation across languages

Results
- We select N relevant benchmarks (mteb(eng), mteb(multilingual), maybe some domains?)

Discussion:
- brief analysis of performance

Discussion/limitations of existing models
- Is there places where we see existing models fail? Language, domains, tasks?

Limitations:
- For large collaborative benchmarks: Continually updating the benchmark invalidated previous model results

Let me know what you guys think?

**Reply by imenelydiaker** [reply]
I took some liberties and organized the overleaf to facilitate the writing. I copied the outline you suggested here 🙂

**Reply by KennethEnevoldsen** [reply]
I have also created an issue #896  to get an overview of all the segment that needs to be written

**Comment by KennethEnevoldsen** [comment]
@Muennighoff, @imenelydiaker I have sent an email to you guys to set up a meeting to discuss this and the next stages of mmteb

**Comment by imenelydiaker** [comment]
EMNLP 2024 deadline (abstract + paper): June 15, 2024 11:59pm AOT ([Wesbiste](https://2024.emnlp.org/))
Template: https://github.com/acl-org/acl-style-files

**Reply by Muennighoff** [reply]
After discussion we realized EMNLP probably makes most sense so swapped to the template: https://www.overleaf.com/8693731561tpntkddrhngs#41bcff Also turned on track changes :)

**Reply by imenelydiaker** [reply]
TACL template in case we don't make it for EMNLP: https://www.overleaf.com/project/61c0608457aa0222e6ced44a

**Comment by isaac-chung** [comment]
Was talking to @imenelydiaker about the TACL [submission guideline](https://transacl.org/ojs/index.php/tacl/about/submissions#authorGuidelines) and realized that:
> The submission is not a version of a paper that was reviewed by the  ACL, AACL, EACL,  NAACL, or EMNLP conferences within the last 9 months. Note that papers that were withdrawn before receiving a decision but nonetheless received reviews are thus not TACL-eligible during the 9-month window.

So if we submit to EMNLP/ACL in June, we won't be able to submit to TACL for 9 months from the review date.

**Comment by imenelydiaker** [comment]
I would appreciate some help on the related work part, I'm missing some references for these points:

1. **Speeding up benchmarks**: Any papers you know of can help on this?
2. **Open-source large effort**: Do you have any papers related to this? I've seen [SeaCrowd](https://github.com/SEACrowd), although I haven't found any paper linked to it. I know that [OpenDevin](https://opendevin.github.io/OpenDevin/) is also a community effort project, although there's no paper for the moment, should we talk about it?

Should we include any other works in this part? References to some included datasets?

**Reply by Muennighoff** [reply]
Happy to help on that - which Overleaf are you using? I think we can reference Aya by Cohere4AI, BLOOM by Big Science etc.
Sara Hooker who led Aya at Cohere has also expressed interest in helping us - if you don't object we can involve her in the paper writing :)

**Reply by imenelydiaker** [reply]
I'm using the one you shared with us. I organized it a bit to ease the writing, but feel free to change that. Thanks for the references, I completely forgot Bloom and Aya 😄 

No problem with Sara Hooker joining and helping on the writing!

**Reply by KennethEnevoldsen** [reply]
Hi @imenelydiaker added a bit more to the general introduction. I plan to also start either on the community effort or 'speeding up the benchmark' segments

**Reply by dokato** [reply]
I worked on Aya too, if there's something I can help with @imenelydiaker give me a shout.

**Reply by KennethEnevoldsen** [reply]
Hi @dokato, we would love a helping hand. You might start out taking a look at the related works section as well as the introduction.

Also if there is segment that you feel should be added which is not currently considered in the paper.

**Reply by KranthiGV** [reply]
Happy to help with writing as well. Any sections you want me to start contributing to?

**Reply by KennethEnevoldsen** [reply]
@KranthiGV if you can create a script to add all the citations to a .bib file and create a table overview of all the tasks with the references (you might take inspiration from https://openreview.net/pdf/f5f1953a9c798ec61bb050e62bc7a94037fd4fab.pdf, potentially making it more dense, e.g. exclude description).

This is to ensure that we sight all the relevant dataset authors

**Reply by isaac-chung** [reply]
The task overview table already exists as `tasks.md`. Adding all bibtex from tasks weeks be great. Backfilling missing bibtex would be even better!

**Reply by KennethEnevoldsen** [reply]
Ahh right yes adding the old bibtex would also be great. Though, there is still some work in converting it to latex such that the dataset authors are given a citation.

**Comment by KennethEnevoldsen** [comment]
I have added the dataset construction section (an outline) as well in the appendix. Please refer relevant authors to fill out this section. Tagging a few people I know have something to add:

- @x-tabdeveloping multilevel clustering and multi clf
- @imenelydiaker cross-lingual tasks
- @Rysias wiki clustering construction (as an example)
- @orionw: instruction retrieval and reranking
- @Muennighoff general
- @awinml would love if you could write a bit about the conversion of legalbench as well

**Reply by KennethEnevoldsen** [reply]
feel free ask let contributors who contributed specific segments know about this as well.

**Reply by isaac-chung** [reply]
Just moving the appendix sections into their own files.

**Reply by x-tabdeveloping** [reply]
@KennethEnevoldsen Do we want a section explaining the methodology involved in the new tasks? Like how we did multilabel classification and hierarchical clustering?

**Reply by KennethEnevoldsen** [reply]
I have already written a general introduction to some of the speed-ups (please take a look at it), this goes into the hierarchical clustering as an example. However I agree that we should add a section on new task types

**Reply by isaac-chung** [reply]
Would you prefer if ClusteringFast to be part of this appendix as well? (B)

**Reply by KennethEnevoldsen** [reply]
@isaac-chung we have a section where we describe all the tasks, I would describe clustering there (probably just clusteringfast)

**Reply by isaac-chung** [reply]
Got it, thanks. From the structure now, I'll add that to B.2

**Comment by y12uc231** [comment]
hi all, I would love to contribute with the writing, is there any section that require some more hands?

**Reply by KennethEnevoldsen** [reply]
If @KranthiGV isn't doing the section you can give that one a go:

> @KranthiGV if you can create a script to add all the citations to a .bib file and create a table overview of all the tasks with the references (you might take inspiration from https://openreview.net/pdf/f5f1953a9c798ec61bb050e62bc7a94037fd4fab.pdf, potentially making it more dense, e.g. exclude description). This is to ensure that we cite all the relevant dataset authors

**Reply by KranthiGV** [reply]
@KennethEnevoldsen 
I'm on it. Will be working on this over the weekend.

**Reply by KennethEnevoldsen** [reply]
@KranthiGV where did this end up?

**Reply by Sakshamrzt** [reply]
@KennethEnevoldsen If this is still open I can take this up.

**Reply by KennethEnevoldsen** [reply]
I believe it is

**Reply by KennethEnevoldsen** [reply]
@Sakshamrzt I have added you to an issue here: #897

**Reply by Sakshamrzt** [reply]
@KennethEnevoldsen can you tell how was the current custom.bib file built?

**Reply by KennethEnevoldsen** [reply]
I believe it is copy and paste  (each contributor just copied their required bibtex in)

**Comment by mariyahendriksen** [comment]
Hi everyone,

I plan to spend some time working on the benchmark this weekend, adding additional datasets and I would also like to contribute to the paper writing. Is there anything specific you would like me to focus on for the paper draft? What would be most helpful?

Thanks!

**Reply by isaac-chung** [reply]
Hey! Amazing to have you. For datasets there's an outstanding issue on converting clustering tasks to a fast format #660 , and (need @imenelydiaker to confirm) the French ones still need to be done  #568 , so would be great if you could take a look at those!

In terms of the paper draft, maybe @KennethEnevoldsen could help guide us here.

**Reply by imenelydiaker** [reply]
Amazing @mariyahendriksen, just as @isaac-chung mentionned it there are some open issues that we need to address quickly for the submission, you can take #568! Happy to have you as a contributor!

**Reply by imenelydiaker** [reply]
For the paper, there is this section to fill in the appendix about _Semantic Text Similarity_ datasets and tasks. You can have a look at it if you want, else the list is here: https://github.com/embeddings-benchmark/mteb/discussions/595#discussioncomment-9444047

**Reply by isaac-chung** [reply]
We've added more "[good first issues](https://github.com/embeddings-benchmark/mteb/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)" as well. Feel free to assign to yourself if you plan to work on it!

**Reply by mariyahendriksen** [reply]
thanks, I will take a look tonight!

**Reply by mariyahendriksen** [reply]
> For the paper, there is this section to fill in the appendix about _Semantic Text Similarity_ datasets and tasks. You can have a look at it if you want, else the list is here: [#595 (comment)](https://github.com/embeddings-benchmark/mteb/discussions/595#discussioncomment-9444047)


Thanks! Done with Semantic Text Similarity, will fill the gaps for Instruction Retrieval and Reranking next.

**Reply by mariyahendriksen** [reply]
done with the Appendix D, what would be another important thing to focus on next?
I am thinking between Conclusions and Appendix D (Full results). 
But perhaps there is something else?

**Reply by KennethEnevoldsen** [reply]
@mariyahendriksen thanks, sorry for the late reply I had to get an overview of paper again before responding (you can see it here #896). I am assuming you are referring to what is now Appendix B. 

Could I ask you to try to split appendix B up into task introduction (describing the old and new tasks) and task construction (describing how the datasets were constructed first generally and allowing for specific cases like Appendix C allowing for more descriptions like it). This would naturally integrate appendix C into task construction.

**Reply by mariyahendriksen** [reply]
@KennethEnevoldsen sure, I will look into it tonight, will keep you posted!

**Reply by mariyahendriksen** [reply]
i split the appendix into two parts as we discussed. 
Anything else I should focus on?

**Reply by KennethEnevoldsen** [reply]
Thanks @mariyahendriksen. I have made a few changes:

1) Converted C to B.3 for new datasets (and C is now an overview table)
2) started differentiating B1 and B2 such that B1 explains the task and B2 explains task construction. I'd like to ask you to finalize this if you have the time (I have done this for some of the tasks already). Feel free to leave the subsection empty if you are not sure how to fill it out.

**Reply by mariyahendriksen** [reply]
@KennethEnevoldsen I have gone over Appendix B one more time, some parts are missing but I see that they were already assigned to someone else so I will leave them as they are for now. If no one picks them up in the upcoming days, I am happy to fill-in the gaps myself.
Regardless, I will keep an eye on the Appendix to make sure all the parts are aligned.

What should focus on now?

**Reply by KennethEnevoldsen** [reply]
@mariyahendriksen I don't have any section atm. will take a look through the paper today - will let you know if something comes up

**Comment by asparius** [comment]
There seems to be a typo, it is M3TEB in the paper but M2TEB in abstract. Anything else I can help with paper writing or any other issue?

**Reply by KennethEnevoldsen** [reply]
@asparius I believe we have agreed on using MMTEB going forward (thought M2TEB and M3TEB were discussed)

**Comment by jaygala24** [comment]
@KennethEnevoldsen I would also like to help with the paper writing. Let me know where can I be of help. Thank you!

**Reply by KennethEnevoldsen** [reply]
@jaygala24 I have created an overview issue here #896. You could take a look at the related works section or limitations

**Comment by mariyahendriksen** [comment]
Did we miss the deadline? What’s our current target date? The next ACL ARR submission window is in August; should we aim for that? Alternatively, we could also consider TMLR with a self-imposed deadline.

**Reply by KennethEnevoldsen** [reply]
@mariyahendriksen I believe the next deadline is a preprint on arxiv this summer (which leaves us open to quite a few venues afterwards).

**Reply by mariyahendriksen** [reply]
ok, sounds good! do you happen to know the exact date of the arxiv deadline? knowing this would really help planning

**Reply by KennethEnevoldsen** [reply]
I do not - the current plan is to get us ready to run models as soon as possible (my hope is to get that started this month), then it would take about ~14 days to run the models. Then, about ~14 days for writing after that, which leaves us at approximately the start of August. 

I will take it up with the steering group though to see if people agree.

**Reply by imenelydiaker** [reply]
I believe TACL is more adapted to our work: https://transacl.org/index.php/tacl, it's like TMLR but for NLP.

**Reply by gentaiscool** [reply]
hi all, I was wondering if we have a new target submission for the paper. Thank you!
